### DISCUSSION TITLE
Single Statement create column from computation

### DISCUSSION THREAD

**Discussion by HiramTheHero** [duckdb/duckdb#6874] [Ideas]
Instead of doing:
```
ALTER TABLE iris ADD COLUMN sum_result FLOAT;
UPDATE iris SET sum_result = SepalLength + SepalWidth;
```
Having something like the following would be nice:
`ALTER TABLE iris ADD COLUMN sum_result INTEGER DEFAULT SepalLength + SepalWidth;`

If the command directly above is attempted, the following error results:

`Error: Binder Error: DEFAULT value cannot contain column names`

**Comment by Tishj** [comment]
If you just want to create a column as an expression that references other columns, we have VIRTUAL GENERATED columns for that.
Have a look at th docs to see if that would cover your needs 👍

**Reply by HiramTheHero** [reply]
Thank you for the response. I did see VIRTUAL GENERATED columns in the documentation. From what I understand in the documentation, it looks like VIRTUAL GENERATED columns are only compatible with a CREATE TABLE statement. Is there another way to use them after a table is created?

The following:
`alter table iris add virtual column sum_result as SepalLength + SepalWidth;`

Throws an error:
```
Error: Parser Error: syntax error at or near "column"
LINE 1: alter table iris add virtual column sum_result as SepalLength + Sepa...
```

**Reply by HiramTheHero** [reply]
I actually got a response about this on the Discord chat.

The code:
`alter table iris add column sum_result generated always as (SepalLength + SepalWidth) virtual;`

Will produce the following output:
`Error: Parser Error: Adding generated columns after table creation is not supported yet`

So, it looks like the feature is, somewhere, in progress.

**Reply by Tishj** [reply]
Ah yea, I don't remember the specifics of this, but I worked on this, so I am confident there was a reason we don't support it yet
